src/main.cpp

```diff
@@ -4591,6 +4591,21 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     }
 
 
+    else if (!(nLocalServices & NODE_BLOOM) &&
+              (strCommand == "filterload" ||
+               strCommand == "filteradd" ||
+               strCommand == "filterclear") &&
```

Comment:
Ages back, before this was merged, I'd written code that permitted turning off TX relaying during IBD (when relaying TXs is less useful to do most TXs being orphans), and then when we were getting close to the latest block, it would send a "filterclear" message to all connected nodes in order for them to start relaying TXs. If I was to be using those code today, I would be getting banned from every fairly recent Bitcoin Core node. Somehow seems a little over-the-top considering all I am effectively doing is offering to relay ALL TXs they send me. Surely there isn't any DoS vector in getting a node to revert to behaviour that has existed since the very first version of the Satoshi client - and given this pull request was primarily for anti-DoS reasons, then could perhaps "filterclear" be excluded from the list of messages that cause the node to be banned? Or is thre a DoS vector in allowng only "filterclear" messages? Without allowing "filterclear" then I can see no other way to enable TX relaying at a point in time after the version message has been sent.